File: 0700_easy_hot_search-in-a-binary-search-tree/solution.py
# Definition for a binary tree node.
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

        
class Solution:
    def searchBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if not root:
            return None
        
        ptr = root

        while ptr:
            if val == ptr.val:
                break
            if val < ptr.val:
                ptr = ptr.left
            else:
                ptr = ptr.right
        
        return ptr

File: 0700_easy_hot_search-in-a-binary-search-tree/solution.java
/**
 * Definition for a binary tree node.
 */

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}
 
 
class Solution {
    public TreeNode searchBST(TreeNode root, int val) {
        if (root == null) {
            return null;
        }

        TreeNode ptr = root;

        while(ptr != null) {
            if(val < ptr.val) {
                ptr = ptr.left;
            } else if (val > ptr.val) {
                ptr = ptr.right;
            } else if (val == ptr.val) {
                break;
            }

        }

        return ptr;
    }
}